assets/aml-benchmark/components/src/prompt_crafter/main.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Entry script for Prompt Crafter Component."""
+import os
+from argparse import ArgumentParser
+import logging
+
+from .package.prompt_crafter import PromptCrafter
+from .package.prompt import PromptType
+from utils.exceptions import swallow_all_exceptions
+from utils.logging import get_logger, log_mlflow_params
+
+
+logger = get_logger(__name__)
+
+
+def parse_args() -> ArgumentParser:
+    """Parse command line arguments."""
+    parser = ArgumentParser()
+    parser.add_argument("--input_dir", type=str, required=False)
+    parser.add_argument("--test_data", type=str, required=True)
+    parser.add_argument("--few_shot_data", type=str, required=False)
+    parser.add_argument("--prompt_type", type=str, required=True,
+                        choices=[PromptType.chat.name, PromptType.completions.name])
+    parser.add_argument("--n_shots", type=int, required=True)
+    parser.add_argument("--random_seed", type=int, default=0, required=True)
+    parser.add_argument("--output_file", type=str, required=True)
+    parser.add_argument("--output_dir", type=str, required=False)
+    parser.add_argument("--output_mltable", type=str, required=False)
+    parser.add_argument("--few_shot_dir", type=str, required=False)
+    parser.add_argument("--input_filename", type=str, required=False)
+    parser.add_argument("--few_shot_filename", type=str, required=False)
+    parser.add_argument("--metadata_keys", type=str, required=False)
+    parser.add_argument("--prompt_pattern", type=str, required=False)
+    parser.add_argument("--few_shot_pattern", type=str, required=False)
+    parser.add_argument("--few_shot_separator", type=str, required=False)
+    parser.add_argument("--prefix", type=str, required=False)
+    parser.add_argument("--label_map", type=str, required=False)
+    parser.add_argument("--system_message", type=str, required=False)
+    parser.add_argument("--output_pattern", type=str, required=False)
+    parser.add_argument("--additional_payload", type=str, required=False)
+
+    return parser.parse_args()
+
+
+@swallow_all_exceptions(logger)
+def main() -> None:
+    """Entry function for Prompt Crafter Component."""
+    args = parse_args()
```

Comment:
We should avoid parsing args inside `main` function. All of the args should be passed to the `main` function as parameters and must have proper documentation. Please check [guidelines](https://github.com/Azure/azureml-assets/blob/main/assets/aml-benchmark/README.md#guidelines) for more info.

Reply:
Updated. Thanks!

Reply:
Updated. Thanks